service/utils.go

```diff
@@ -21,7 +21,10 @@ const (
 func getEnvironmentId(r *http.Request) (string, error) {
 	environment := r.Header.Get(environmentIdHeader)
 	if environment == "" {
-		return "", fmt.Errorf("Request is missing environment header %s", environment)
+		environment = r.URL.Query().Get("projectId")
+		if environment == "" {
+			return "", fmt.Errorf("Request is missing environment header %s", environment)
```

Comment:
Error message is no longer accurate, since environment can come from header or query string parameter. `Request is missing environment id" would suffice. Also, it doesn't make sense to include the environment variable in this error message, since it is guaranteed to be blank (I realize it was that way before your change)